spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/SpanCloser.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.web;
+
+import java.lang.invoke.MethodHandles;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.SpanReporter;
+import org.springframework.cloud.sleuth.Tracer;
+import org.springframework.http.HttpStatus;
+
+/**
+ * A helper class to close spans
+ *
+ * @author Marcin Grzejszczak
+ */
+class SpanCloser {
+
+	private static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
+
+	private Tracer tracer;
+	private SpanReporter spanReporter;
+	// we want to share the logic of adding response tags between interceptor and filter
+	// also the add response tags method is protected and we want it to be consistent between
+	// filter / interceptors
+	private TraceFilter traceFilter;
+	private BeanFactory beanFactory;
+
+	SpanCloser(Tracer tracer, SpanReporter spanReporter, TraceFilter traceFilter) {
+		this.tracer = tracer;
+		this.spanReporter = spanReporter;
+		this.traceFilter = traceFilter;
+	}
+
+	SpanCloser(BeanFactory beanFactory) {
+		this.beanFactory = beanFactory;
+	}
+
+	void detachOrCloseSpans(HttpServletRequest request,
+			HttpServletResponse response, Span spanFromRequest, Throwable exception) {
+		Span span = spanFromRequest;
+		if (log.isDebugEnabled()) {
+			log.debug("Will try to detach or close " + span + "");
+		}
+		if (span != null) {
```

Comment:
I know this code is mostly moved, but if you early return when span is null, it saves a level of indent